const main = 
  [
    {id:1,name:"John"}, 
    {id:2,name:"James"},
    {id:3,name:"Jim"}
  ]

const app = () => {
  
}

let fetchSomeData = New promise((resolve, reject) => {
  if (!allGood) {
    reject("error")
  } else {
    resolve {
      id:1,
        message: "complete

    });

mainData.fetch(response => {
}).then(data => {
  console.log(data)
})
    
    
fetchSomeData.then(fetchedData => {
  console.log('then', fetchData)
}).catch(err => {
  console.error('catch',err);
})
